packages/eslint-plugin-template/src/rules/i18n.ts

```diff
@@ -183,17 +191,22 @@ export default createESLintRule<Options, MessageIds>({
     const allowedTags: ReadonlySet<string> = new Set(ignoreTags);
     const collectedCustomIds = new Map<string, readonly ParseSourceSpan[]>();
 
-    function handleElement({
-      i18n: { customId },
-      name,
-      parent,
-      sourceSpan,
-    }: StronglyTypedElement) {
-      if (allowedTags.has(name) || isElementWithI18n(parent)) {
+    function handleElementOrTemplate(
+      node: StronglyTypedElement | StronglyTypedTemplate,
+    ) {
+      const { i18n, parent, sourceSpan } = node;
+
+      if (
+        isTagAllowed(allowedTags, node) ||
+        isElementWithI18n(parent) ||
+        isTemplateWithI18n(parent)
+      ) {
         return;
       }
 
-      if (!isEmpty(customId)) {
+      const customId = getI18nCustomId(i18n);
+
+      if (customId) {
```

Comment:
This was one of the causes of having `Cannot read property 'start' of undefined` related in #701 (this message comes from `ESLint` [here](https://github.com/eslint/eslint/blob/0a1a850575ca75db017051abe5e931f0f9c8012b/lib/linter/linter.js#L1236) — as the `Node` doesn't contain a property called `loc` it tries to access `undefined.start` and thus the error doesn't bubble up correctly).

---

Actually `customId` can be `undefined` in some cases, but we were using `Message` when we should use `I18nMeta` instead and check it properly.